webapp/src/app.tsx

```diff
@@ -1335,7 +1335,7 @@ export class ProjectView
                                 // We are too late; the editor has already been loaded.
                                 // Call the onChanges handler to update the editor.
                                 pxt.tickEvent(`identity.syncOnProjectOpen.timedout`, { 'elapsedSec': elapsed})
-                                if (changes.length)
+                                if (changes.some(header => header.id === h.id))
```

Comment:
@darzu does this make sense? I was getting an infinite reload and this fixed it. The current project wasn't in `changes`, yet it was still reloading the editor.

Reply:
Ahh I see, good catch.
A few lines up, you can see cloud.syncAsync only passes a single header, h, the one that we're just loading. So I thought it'd be safe to just check length here.
But it's not because the way cloud.syncAsync works is that if a sync is already in progress it just returns a promise to that one which might include more headers than the one you asked for.

 (Really the whole syncAsync thing needs to be redone.. which I've done in my refactor branch but that's not coming in yet.)